src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -122,6 +124,50 @@ function matrix(A::Array{AbstractAlgebra.Generic.FreeModuleElem{T},1}) where T <
    return X
 end
 
+"""
+    upper_triangular_matrix(L::AbstractVector{T}) where T <: RingElem
+
+Return the upper triangular matrix whose non-zero entries (ordered lexicographically) are the elements of the sequence `L`.
+
+An error is returned whenever the length of `L` has not the form `n*(n+1)/2` for some integer `n`.
+"""
+function upper_triangular_matrix(L::AbstractVector{T}) where T <: RingElem
+   d = 1                # dimension of the matrix
+   while div(d*(d+1),2)< length(L)
+      d+=1
+   end
```

Comment:
`d*(d+1)/2 == length(L)` => `(2d+1)^2 == 4d^2 + 4d + 1 == 8*length(L)+1` => `d = (sqrt(8*length(L)+1)-1)/2`
So modulo mistakes I made...
```suggestion
   d = (sqrt(8*length(L)+1)-1)/2
```